src/persist-client/src/internal/machine.rs

```diff
@@ -361,11 +361,36 @@ where
         let mut machine = self;
         spawn(|| "persist::heartbeat_read", async move {
             let sleep_duration = machine.cfg.reader_lease_duration / 2;
+            let mut last_wakeup = Instant::now();
             loop {
                 tokio::time::sleep(sleep_duration).await;
+                let wakeup = Instant::now();
+
+                let elapsed_since_last_wakeup = wakeup - last_wakeup;
+                if elapsed_since_last_wakeup > machine.cfg.reader_lease_duration {
```

Comment:
I'd like for this to trigger on something like `elapsed > sleep_duration + Duration::from_secs(60)`